### PR TITLE
fix(Avatar): optimised images

### LINKED ISSUE #6651: Author images are very large on DCR fronts
This happens on every card that has an author image, but is easier to see on the [contributor index fronts](https://www.theguardian.com/index/contributors).

On frontend the image used is:
https://i.guim.co.uk/img/uploads/2017/10/09/Zoe-Williams,-L.png?width=216&quality=85&auto=format&fit=max&s=ba60aca1be463470a65bb6b768b6c457

On DCR the image used is:
https://uploads.guim.co.uk/2017/10/09/Zoe-Williams,-L.png

We should be optimising the image as we do with other images.

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Augment the `Avatar` component with optimised image widths.

## Why?

Closes #6651

## Screenshots

It’s visually identical. The new images are smaller in byte size, though!

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |
| [George-Monbiot,-L.png](https://uploads.guim.co.uk/2017/10/06/George-Monbiot,-L.png) | [George-Monbiot,-L.png?width=140&quality=45&dpr=2&s=none](https://i.guim.co.uk/img/uploads/2017/10/06/George-Monbiot,-L.png?width=140&quality=45&dpr=2&s=none) |

[before]: https://user-images.githubusercontent.com/76776/205667312-29fb9fc5-87c4-4412-b67e-45cfd42367cb.png
[after]: https://user-images.githubusercontent.com/76776/205667234-3d07a4b1-8119-4233-a742-87350c555e8f.png

### CODE REVIEW

File: dotcom-rendering/src/web/components/Avatar.tsx [hunk 1]
```diff
@@ -1,39 +1,107 @@
...
+	if (url.hostname !== 'i.guim.co.uk') return src;
+
+	const [, , domain] = url.pathname.split('/');
+	if (!['uploads', 'media'].includes(domain)) return src;
+
+	return new URL(
+		url.pathname.replace(`/img/${domain}/`, ''),
+		`https://${domain}.guim.co.uk`,
+	).toString();
+};
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Avatar.tsx:63]: This is a bit unfortunate, but we had to have a similar approach when we first introduced `CardPicture`, as the image paths coming from `frontend` are already signed.

File: dotcom-rendering/src/web/lib/getSourceImageUrl_temp_fix.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *   `https://media.guim.co.uk/img/media/c56c6217f8fa306d89320cc7915161203b5017b1/0_166_5000_3002/master/5000.jpg`
+ *
+ * - from
+ *   `https://i.guim.co.uk/img/uploads/2017/10/06/George-Monbiot,-L.png?width=173&quality=85&auto=format&fit=max&s=be5b0d3f3aa55682e4930057fc3929a3`
+ *   to
+ *   `https://uploads.guim.co.uk/2017/10/06/George-Monbiot,-L.png`
+ *
+ * @deprecated Favour updating image paths sent from frontend to sources
+ */
+export const getSourceImageUrl = (source: string): string => {
```
Comment (AshCorr) [inline dotcom-rendering/src/web/lib/getSourceImageUrl_temp_fix.ts:22]: Still needed? Looking at a front json the images used by contributors seem to be using uploads.guim.co.uk now, but haven't looked too deeply.

**Review by AshCorr — APPROVED** [review]

File: dotcom-rendering/src/web/components/Avatar.tsx [hunk 3]
```diff
@@ -1,39 +1,86 @@
...
+
+	/**
+	 * The assumption here is readers on devices that do not support srcset
+	 * are likely to be on poor network connections so we're going
+	 * to fallback to the smallest image at low resolution.
+	 *
+	 * Sources are ordered in `descendingByBreakpoint` order,
+	 * so the last one is the smallest.
+	 */
+	const [{ lowResUrl: fallbackSource }] = sources.slice(-1);
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/src/web/components/Avatar.tsx:59]: Ah yes I should have rebased against yours.

File: dotcom-rendering/src/web/lib/getSourceImageUrl_temp_fix.ts (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/lib/getSourceImageUrl_temp_fix.ts:22]: There were loads of errors in Chromatic, and we still have it for Carousel cards as #6307 was closed unmerged.

**Pull request merged**